### DISCUSSION TITLE
Decimal, numeric field overflow when seeding. (v2.23.0)

### DISCUSSION THREAD

**Discussion by l2D** [prisma/prisma#7199] [Q&A] [answered]
Hello,

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; I tried to seed the Lat/Lng in Postgres but I got some errors about precision and scale. At first, I set to ``` @db.Decimal(8,6) ``` and got this error. 

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;After that, I changed ``` @db.Decimal(8,6) ``` to ``` @db.Decimal(12,7) ``` and run ``` npx prisma db push``` and ```npx prisma migrate reset ``` 
but still not working then I check the database in ``` location_lat && location_lng ``` on ``` Restaurants table ```. The type of these are ``` numeric ``` (precision: 12, scale: 7) and after changed on database to ``` decimal ```(precision: 12, scale: 7) still not work.

Error:
<pre><code>PrismaClientUnknownRequestError2 [PrismaClientUnknownRequestError]:
Invalid `prisma.restaurants.create()` invocation:

  Error occurred during query execution:
ConnectorError(ConnectorError { user_facing_error: None, kind: QueryError(Error { kind: Db, cause: Some(DbError { severity: "ERROR", parsed_severity: Some(Error), code: SqlState("22003"), message: "numeric field overflow", detail: Some("A field with precision 8, scale 6 must round to an absolute value less than 10^2."), hint: None, position: None, where_: None, schema: None, table: None, column: None, datatype: None, constraint: None, file: Some("numeric.c"), line: Some(6612), routine: Some("apply_typmod") }) }) })</code></pre>

Schema:
```javascript
model Restaurants {
  restaurant_id Int       @id @default(autoincrement())
  merchant_id   Int
  name          String
  image_url     String
  products      Json
  location_lat  Decimal   @db.Decimal(12, 7)
  location_lng  Decimal   @db.Decimal(12, 7)
  open_hour     String
  close_hour    String
  open_days     Json
  created_at    DateTime  @default(now())
  updated_at    DateTime  @updatedAt
  delete_flag   Boolean   @default(false)
  Merchant      Merchants @relation(fields: [merchant_id], references: [merchant_id])
  Jobs          Jobs[]
}
```

<details>
<summary>Seed:</summary>
<pre><code>
const restaurantsData = [
  {
    Merchant: {
      connect: { merchant_id: 1 },
    },
    name: "[Name1]",
    image_url: "PATH",
    products: [
      {
        product_id: 1,
        product_name: "[Food 1]",
        product_price: 50,
        product_discount: 0,
        product_image_url: "PATH",
        product_status: "Available",
      },
      {
        product_id: 2,
        product_name: "[Food 2]",
        product_price: 60,
        product_discount: 20,
        product_image_url: "PATH",
        product_status: "Available",
      },
      {
        product_id: 3,
        product_name: "[Food 3]",
        product_price: 60,
        product_discount: 0,
        product_image_url: "PATH",
        product_status: "Unavailable",
      },
    ],
    location_lat: 13.8649955,
    location_lng: 100.5196122,
    open_hour: "10.00 - 21.00",
    close_hour: "21.00 - 10.00",
    open_days: {
      Mon: false,
      Tue: true,
      Wed: true,
      Thu: true,
      Fri: true,
      Sat: true,
      Sun: true,
    },
  },
];

async function main() {
    for (const data of restaurantsData) {
         const restaurant = await prisma.restaurants.create({
            data: data,
          });
}

main()
  .catch((e) => {
    console.error(e);
    process.exit(1);
  })
  .finally(async () => {
    await prisma.$disconnect();
  });

</code></pre></details>

**Comment by ryands17** [comment, marked as answer]
@l2D 👋 

I tried your schema with the required fields:

```prisma
model Restaurants {
  restaurant_id Int      @id @default(autoincrement())
  name          String
  image_url     String
  location_lat  Decimal  @db.Decimal(12, 7)
  location_lng  Decimal  @db.Decimal(12, 7)
  open_hour     String
  close_hour    String
  open_days     Json
  created_at    DateTime @default(now())
  updated_at    DateTime @updatedAt
  delete_flag   Boolean  @default(false)
}
```

And ran the following seed script to check if the values returned are correct:

```ts
const restaurantsData = [
  {
    name: '[Name3]',
    image_url: 'PATH',
    location_lat: 13.8649955,
    location_lng: 100.5196122,
    open_hour: '10.00 - 21.00',
    close_hour: '21.00 - 10.00',
    open_days: {
      Mon: false,
      Tue: true,
      Wed: true,
      Thu: true,
      Fri: true,
      Sat: true,
      Sun: true,
    },
  },
]

async function main() {
  for (let data of restaurantsData) await prisma.restaurants.create({ data })
  console.log(await prisma.restaurants.findMany())
}
```

I got a successful response:

![image](https://user-images.githubusercontent.com/19697099/119131568-435cac00-ba57-11eb-9e10-e2518c7ffc1e.png)

I would suggest trying this on a fresh database by running `prisma migrate dev` and then executing the seed script.

**Reply by l2D** [reply]
Thank you for your help. After I run ``` npx prisma migrate dev ``` and ``` npx prisma migrate reset ```, It's working correctly.
